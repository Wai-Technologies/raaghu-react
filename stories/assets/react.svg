<svg width="84" height="80" viewBox="0 0 84 80" fill="none" xmlns="http://www.w3.org/2000/svg">
<path d="M41.9998 46.3032C45.3591 46.3032 48.0824 43.54 48.0824 40.1315C48.0824 36.7229 45.3591 33.9597 41.9998 33.9597C38.6404 33.9597 35.9171 36.7229 35.9171 40.1315C35.9171 43.54 38.6404 46.3032 41.9998 46.3032Z" fill="#61DAFB"/>
<path d="M41.9998 46.3032C45.3591 46.3032 48.0824 43.54 48.0824 40.1315C48.0824 36.7229 45.3591 33.9597 41.9998 33.9597C38.6404 33.9597 35.9171 36.7229 35.9171 40.1315C35.9171 43.54 38.6404 46.3032 41.9998 46.3032Z" stroke="#61DAFB" stroke-width="4"/>
<path d="M42 55.4964C64.0914 55.4964 82 48.5584 82 40C82 31.4415 64.0914 24.5035 42 24.5035C19.9086 24.5035 2 31.4415 2 40C2 48.5584 19.9086 55.4964 42 55.4964Z" stroke="#61DAFB" stroke-width="4"/>
<path d="M28.7734 47.7482C39.8191 67.1601 54.6952 79.4276 62 75.1484C69.3048 70.8692 66.2723 51.6637 55.2266 32.2518C44.1809 12.8398 29.3048 0.572317 22 4.85154C14.6952 9.13076 17.7277 28.3363 28.7734 47.7482Z" stroke="#61DAFB" stroke-width="4"/>
<path d="M28.7732 32.2518C17.7275 51.6637 14.695 70.8692 21.9998 75.1485C29.3046 79.4277 44.1807 67.1602 55.2264 47.7482C66.2721 28.3363 69.3046 9.1308 61.9998 4.85158C54.695 0.572351 39.8189 12.8398 28.7732 32.2518Z" stroke="#61DAFB" stroke-width="4"/>
</svg>
