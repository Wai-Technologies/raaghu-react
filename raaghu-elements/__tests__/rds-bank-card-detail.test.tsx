import React from "react";
import "@testing-library/jest-dom";
import { render, fireEvent } from "@testing-library/react";
import RdsBankCardDetail, { RdsBankCardDetailProps } from "../src/rds-bank-card-detail/rds-bank-card-detail";

jest.mock('react-lottie-player', () => ({
    __esModule: true,
    default: jest.fn(),
  }));


const mockCardDatas = [
    {
        icon: "card-icon",
        iconHeight: 24,
        iconWidth: 24,
        iconFill: "#000000",
        iconStroke: "#FFFFFF",
        iconColorVariant: "primary",
        cardName: "Test Card 1",
        cardNumber: "1234",
        cardExpiry: "12/23",
    },
    {
        icon: "card-icon",
        iconHeight: 24,
        iconWidth: 24,
        iconFill: "#000000",
        iconStroke: "#FFFFFF",
        iconColorVariant: "secondary",
        cardName: "Test Card 2",
        cardNumber: "5678",
        cardExpiry: "09/24",
    },
];

const defaultProps: RdsBankCardDetailProps = {
    cardDatas: mockCardDatas,
    isSelectable: true,
    isEditable: true,
};

describe("RdsBankCardDetail", () => {
    const cardDatas = [
        {
            icon: "icon-1",
            iconHeight: 24,
            iconWidth: 24,
            iconFill: "#000",
            iconstroke: "#000",
            iconColorVarient: "primary",
            cardName: "Card 1",
            cardNumber: "1234",
            cardExpiry: "12/23",
        },
        {
            icon: "icon-2",
            iconHeight: 24,
            iconWidth: 24,
            iconFill: "#000",
            iconstroke: "#000",
            iconColorVarient: "secondary",
            cardName: "Card 2",
            cardNumber: "5678",
            cardExpiry: "10/25",
        },
    ];
    
    it("renders correctly", () => {
        const { container } = render(<RdsBankCardDetail {...defaultProps} />);
        expect(container).toMatchSnapshot();
    });


    it("does not render the edit link when not editable", () => {
        const { queryByText } = render(
            <RdsBankCardDetail {...defaultProps} isEditable={false} />
        );
        const editLink = queryByText("Edit");

        expect(editLink).not.toBeInTheDocument();
    });
    it("does not render edit link when not editable", () => {
        const { queryByText } = render(
            <RdsBankCardDetail cardDatas={cardDatas} isSelectable={true} isEditable={false} />
        );

        expect(queryByText("Edit")).toBeNull();
    });

    it("renders the radio button when selectable", () => {
        const { getAllByRole } = render(<RdsBankCardDetail {...defaultProps} />);
        const radioButtons = getAllByRole("radio");

    }); });
