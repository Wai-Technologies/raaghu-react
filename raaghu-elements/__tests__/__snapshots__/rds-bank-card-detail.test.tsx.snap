// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RdsBankCardDetail renders correctly 1`] = `
<div>
  <div
    class="m-1 p-1 "
  >
    <div>
      <div
        class=" mb-4"
      >
        <div
          class="border__gray p-4"
        >
          <div
            class=" "
          >
            <div
              class="d-flex justify-content-between"
            >
              <div
                class="d-flex  align-items-center"
              >
                <span
                  class="cursor-pointer undefined"
                  role="img"
                />
                <div
                  class="ms-3  "
                >
                  <div
                    class="ms-3 d-flex fs-6"
                  >
                    <span>
                      Test Card 1
                    </span>
                    <span
                      class="ms-1 me-1 "
                    >
                      Ending with
                    </span>
                    <span>
                      1234
                    </span>
                  </div>
                  <div
                    class=" ms-3 fs-7"
                  >
                    <span
                      class="text-muted fs-7"
                    >
                       
                      Expiry 
                      12/23
                    </span>
                  </div>
                </div>
              </div>
              <div
                class="form-check"
              >
                <input
                  class="form-check-input"
                  id="flexRadioD"
                  name="flexRadioDefault"
                  type="radio"
                />
              </div>
            </div>
          </div>
          <div
            class="mt-2  ms-5"
          >
            <a
              class="ms-3 text-primary text-decoration-none "
            >
              Set as default
            </a>
            <a
              class="text-primary ms-2 text-decoration-none "
              style="text-decoration: auto;"
            >
              Edit
            </a>
          </div>
        </div>
      </div>
    </div>
    <div>
      <div
        class=" mb-4"
      >
        <div
          class="border__gray p-4"
        >
          <div
            class=" "
          >
            <div
              class="d-flex justify-content-between"
            >
              <div
                class="d-flex  align-items-center"
              >
                <span
                  class="cursor-pointer undefined"
                  role="img"
                />
                <div
                  class="ms-3  "
                >
                  <div
                    class="ms-3 d-flex fs-6"
                  >
                    <span>
                      Test Card 2
                    </span>
                    <span
                      class="ms-1 me-1 "
                    >
                      Ending with
                    </span>
                    <span>
                      5678
                    </span>
                  </div>
                  <div
                    class=" ms-3 fs-7"
                  >
                    <span
                      class="text-muted fs-7"
                    >
                       
                      Expiry 
                      09/24
                    </span>
                  </div>
                </div>
              </div>
              <div
                class="form-check"
              >
                <input
                  class="form-check-input"
                  id="flexRadioD"
                  name="flexRadioDefault"
                  type="radio"
                />
              </div>
            </div>
          </div>
          <div
            class="mt-2  ms-5"
          >
            <a
              class="ms-3 text-primary text-decoration-none "
            >
              Set as default
            </a>
            <a
              class="text-primary ms-2 text-decoration-none "
              style="text-decoration: auto;"
            >
              Edit
            </a>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
